### PR TITLE
Rename MASTER_DATA_DIRECTORY variable to COORDINATOR_DATA_DIRECTORY

### PR DESCRIPTION
**Pull request by jmcatamney**
The main thrust of this PR is renaming MASTER_DATA_DIRECTORY to COORDINATOR_DATA_DIRECTORY in the utilities code while maintaining backwards compatibility.  As of this PR, any tests still using MASTER_DATA_DIRECTORY will still work if they use the demo cluster (because those scripts now set both variables) or do their own environmental setup; note, however, that COORDINATOR_DATA_DIRECTORY takes precedence over MASTER_DATA_DIRECTORY, so any tests that modify MASTER_DATA_DIRECTORY mid-test while COORDINATOR_DATA_DIRECTORY is set might see some conflicts from the utilities as they will only honor the latter.

It also renames three gpinitsystem variables (MASTER_PORT, MASTER_DIRECTORY, and MASTER_HOSTNAME) with the same backwards compatibility goal and makes a few more master-naming-related fixes that were missed in the previous renaming commit.

When reviewing, the most helpful thing to look out for would be mistaken renames (e.g. `POSTMASTER_` to `POSTCOORDINATOR_`).  The pipeline for this PR is [here](https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/rename_mdd_variable).

### CODE REVIEW

File: gpAux/gpdemo/demo_cluster.sh [hunk 1]
```diff
@@ -286,12 +286,12 @@ cat >> $CLUSTER_CONFIG <<-EOF
 	declare -a DATA_DIRECTORY=(${PRIMARY_DIRS_LIST})
```
Comment (dkrieger-pvtl) [inline gpAux/gpdemo/demo_cluster.sh:286]: L277 above...this wasn't introduced in this PR, but should't 

```
# This names the data directories for the Segment Instances and the Entry Postcoordinator
```

be

```
# This names the data directories for the Segment Instances and the Entry Postmaster
```

File: gpAux/gpdemo/demo_cluster.sh [hunk 2]
```diff
@@ -192,12 +192,12 @@ cat <<-EOF
...
 
-	    GPHOME ................. : ${GPHOME}
-	    MASTER_DATA_DIRECTORY .. : ${QDDIR}/${SEG_PREFIX}-1
+	    GPHOME ................... : ${GPHOME}
+	    COORDINATOR_DATA_DIRECTORY : ${QDDIR}/${SEG_PREFIX}-1
 
-	    MASTER PORT (PGPORT) ... : ${COORDINATOR_DEMO_PORT}
-	    STANDBY PORT ........... : ${STANDBY_DEMO_PORT}
-	    SEGMENT PORTS .......... : ${DEMO_SEG_PORTS_LIST}
+	    MASTER PORT (PGPORT) ..... : ${COORDINATOR_DEMO_PORT}
```
Comment (dkrieger-pvtl) [inline gpAux/gpdemo/demo_cluster.sh:198]: should this read `COORDINATOR PORT`?

**Review by dkrieger-pvtl — APPROVED** [review]
LGTM in the sense that there's nothing obviously wrong.  With a change like this, we need to rely on our tests to catch things.  Though I did find a few nits based on grepping for various words.  Though I didn't find any misspellings of coordinator.

Feel  free to push this now and correct the nits as a follow-up.

File: gpAux/gpdemo/demo_cluster.sh (hunk 2 above)
Comment (jmcatamney) [reply gpAux/gpdemo/demo_cluster.sh]: Good catch, fixed.

File: gpAux/gpdemo/demo_cluster.sh (hunk 1 above)
Comment (jmcatamney) [reply gpAux/gpdemo/demo_cluster.sh:286]: Yep, I'll fix that.

**Pull request merged**